### PR TITLE
Rabbitmq node only reports metrics of itself by default

### LINKED ISSUE #6556: [Metricbeat] RabbitMQ cluster - node metrics reported multiple times (duplicates)
For a RabbitMQ cluster, metricbeat sends 3 beats for each node instead of just one thus triplicating the number of metrics which is unnecessary.

The issue in details is described [here](https://discuss.elastic.co/t/rabbitmq-cluster-node-metrics-reported-multiple-times/123592)

- Version: 6.2.2
- Operating System: Linux Ubuntu 16.04
- Steps to Reproduce: Install metricbeat on all nodes of RabbitMQ cluster and check the records in Elasticsearch.

### PR DESCRIPTION
**Pull request by jsoriano**
A configuration option is added, `node.collect` that can be set to
`cluster` to have the previous behaviour. The default is `node`.

Fixes #6556

### CODE REVIEW

File: metricbeat/module/rabbitmq/node/node.go [hunk 1]
```diff
@@ -40,20 +50,61 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+		BaseMetricSet: base,
+		HTTP:          http,
+		nodesURL:      nodesURL,
+		overview:      overview,
 	}, nil
 }
 
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
-	content, err := m.HTTP.FetchContent()
+type apiOverview struct {
```
Comment (ruflin) [inline metricbeat/module/rabbitmq/node/node.go]: We normally keep all the mapping code in `data.json`. But happy to change this convention :-)

**Comment by jsoriano** [comment]
Ok, agree to allow the user to still collect metrics from all the nodes.

**Comment by jsoriano** [comment]
Added `node.mode` config flag to choose between `self` (new default) and `all` (previous behaviour), to collect metrics only from the node we are connecting to, or from the whole cluster.

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -86,6 +86,7 @@ https://github.com/elastic/beats/compare/v6.0.0-beta2...master[Check the HEAD di
 - Don't stop Metricbeat if aerospike server is down. {pull}6874[6874]
 - disk reads and write count metrics in RabbitMQ queue metricset made optional. {issue}6876[6876]
 - Add mapping for docker metrics per cpu. {pull}6843[6843]
+- RabbitMQ node metricset only collects metrics of the instance it connects to, `node.mode: all` can be used to collect all nodes as before. {issue}6556[6556] {pull}6971[6971]
```
Comment (jsoriano) [inline CHANGELOG.asciidoc]: I wonder if we should move it to breaking changes.

File: metricbeat/module/rabbitmq/node/_meta/docs.asciidoc [hunk 3]
```diff
@@ -1 +1,10 @@
 This is the `node` metricset of the RabbitMQ module.
+
+It collects metrics of nodes, it has two modes that can be selected with the
+`node.mode` setting:
+
+* `self`: collects metrics only from the node from whose endpoint `metricbeat`
+  is collecting metrics. This is the default, and the recommended mode when
+  `metricbeat` is deployed in the same node as the RabbitMQ server.
+* `all`: collects metrics from all the nodes. This is recommended when
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply metricbeat/module/rabbitmq/node/_meta/docs.asciidoc]: Ok, changing to `node.collect` with values `node` and `cluster`.

**Comment by jsoriano** [comment]
Implemented using reporter V2 api now.

File: metricbeat/module/rabbitmq/node/data.go [hunk 4]
```diff
@@ -129,27 +128,22 @@ var (
...
+		eventMapping(r, node)
 	}
-
-	return events, errors
 }
 
-func eventMapping(node map[string]interface{}) (common.MapStr, *s.Errors) {
-	return schema.Apply(node)
+func eventMapping(r mb.ReporterV2, node map[string]interface{}) {
+	event, _ := schema.Apply(node)
```
Comment (jsoriano) [inline metricbeat/module/rabbitmq/node/data.go]: I tried not to ignore errors here, but `Apply` is returning an error that seems like an empty list of missing fields (Literally: "`Required fields are missing: `"). This happens also with the existing code, so this is not a regression. I leave investigating this for a future PR.

Comment (ruflin) [reply metricbeat/module/rabbitmq/node/data.go]: Yes, we need a better general approach here. LGTM for now. Check out my recent changes in the elasticsearch module. There I use the errors for testing purpose only.

**Comment by ruflin** [comment]
jenkins, test this

**Pull request merged**